### PR TITLE
[14.0] [mig] sale_order_ubl

### PR DESCRIPTION
**Pull request by fshah-initos**

### CODE REVIEW

File: sale_order_ubl/models/report.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        pdf_content = super()._post_pdf(
+            save_in_attachment, pdf_content=pdf_content, res_ids=res_ids
+        )
+        if res_ids and len(res_ids) == 1:
+            if self.is_ubl_xml_to_embed_in_sale_order():
+                sale_order = self.env["sale.order"].browse(res_ids)
+                pdf_content = sale_order.embed_ubl_xml_in_pdf(pdf_content)
+        return pdf_content
+
+    def render_qweb_pdf(self, res_ids=None, data=None):
```
Comment (rconjour) [inline sale_order_ubl/models/report.py]: If i'm right, the upstream name of this element is `_render_qweb_pdf`

File: sale_order_ubl/tests/test_ubl_generate.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        buo = self.env["base.ubl"]
+        quotation_states = soo.get_quotation_states()
+        order_states = soo.get_order_states()
+        for i in range(8):
+            i += 1
+            order = self.env.ref("sale.sale_order_%d" % i)
+            for version in ["2.0", "2.1"]:
+                pdf_file = ro.with_context(
+                    ubl_version=version, force_report_rendering=True
+                ).render_qweb_pdf(order.ids)[0]
```
Comment (rconjour) [inline sale_order_ubl/tests/test_ubl_generate.py]: This should also be`_render_qweb_pdf`

**Review by rconjour — CHANGES_REQUESTED** [review]

**Review by rconjour — APPROVED** [review]

**Review by StefanRijnhart — APPROVED** [review]
Thanks!

/ocabot merge nobump

**Comment by StefanRijnhart** [comment]
Bot? Bot? Where are you bot?

/ocabot merge nobump

**Pull request merged**